tests/components/sensor/test_template.py

```diff
@@ -0,0 +1,44 @@
+"""
+tests.components.sensor.template
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tests template sensor.
+"""
+from unittest.mock import patch
+
+import pytest
+
+import homeassistant.core as ha
+import homeassistant.components.sensor as sensor
+
+
+class TestSensorYr:
+    """ Test the Yr sensor. """
+
+    def setup_method(self, method):
+        self.hass = ha.HomeAssistant()
+
+    def teardown_method(self, method):
+        """ Stop down stuff we started. """
+        self.hass.stop()
+
+    def test_template(self, betamax_session):
+        assert sensor.setup(self.hass, {
+            'sensor': {
+                'platform': 'template',
+                'sensors': {
+                    'test_template_sensor': {
+                        'value_template':
+                            '{{ states.sensor.test_state.state }}'
+                    }
+                }
+            }
+        })
+
+        state = self.hass.states.get('sensor.test_template_sensor')
+        assert state.state == ''
```

Comment:
Looks like the template module isn't set up to throw and exception where it can't replace a simple value. This could be changed by adding the following to the exception code, but this breaks some current tests - and perhaps more! I can add an issue for someone to look at this if you like. 
````
ENV.undefined = jinja2.StrictUndefined()
````
I'll revise the test to exercises the ValueException (I think doing some logic with an undefined value will do this).